### PR TITLE
Feature/winrm classes module

### PR DESCRIPTION
**Pull request by bishopbm1**
Added all classes needed to configure WinRM on Windows systems.

We are not using windows_firewall module due to issues maintaining the state of the firewall rules. Will revisit later after updates are made to module.

Classes:
allow_unencrypted
ps_remoting
execution_policy
localaccounttokenfilter
auth
listener::http
listener::https
firewall

Added tests for all classes with 100% code coverage:
```
Finished in 20.92 seconds (files took 1 minute 19.21 seconds to load)
120 examples, 0 failures

.Total resources:   22
Touched resources: 22
Resource coverage: 100.00%
```

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Contains all items to configure WinRM as well as passing variables to the methods.
+class winrm::config {
+
+  assert_private()
+
+  $allow_unencrypted_enable                 = $winrm::params::allow_unencrypted_enable
```
Comment (nmaludy) [inline manifests/config.pp:6]: curios why you're getting all of these parameters here? they shouldn't be needed

File: manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# == Class: winrm
+#
```
Comment (nmaludy) [inline manifests/init.pp:2]: this looks like the "old" way of documenting, not the "puppet strings" way, might look into changing this

File: manifests/init.pp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+class winrm (
+  Boolean  $allow_unencrypted_enable                 = $winrm::params::allow_unencrypted_enable,
+  Boolean  $auth_basic_enable                        = $winrm::params::auth_basic_enable,
+  Boolean  $auth_credssp_enable                      = $winrm::params::auth_credssp_enable,
+  Boolean  $auth_kerberos_enable                     = $winrm::params::auth_kerberos_enable,
+  Boolean  $auth_negotiate_enable                    = $winrm::params::auth_negotiate_enable,
+  String   $certificate_hash                         = $winrm::params::certificate_hash,
+  Integer  $cert_validity_days                       = $winrm::params::cert_validity_days,
+  Enum['AllSigned', 'Bypass', 'RemoteSigned', 'Restricted', 'Undefined', 'Unrestricted']
```
Comment (nmaludy) [inline manifests/init.pp:55]: if you didn't want to copy/past this enum around, you could potentially create your own type alias

File: manifests/service.pp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# License::   MIT
+
+# == Class winrm::service
+#
+# This class is meant to be called from `winrm`
+# It ensure the service is running
+#
+class winrm::service {
+
+  assert_private()
```
Comment (nmaludy) [inline manifests/service.pp:12]: the assert private is probably not needed here and might hurt someone if they try to use this in the future

File: manifests/init.pp (hunk 3 above)
Comment (bishopbm1) [reply manifests/init.pp:55]: This is the only place i have the Enum. I thought by placing it here in the init that would be the only place i needed it?

Comment (nmaludy) [reply manifests/init.pp:55]: Yeah that's fine, was just general comment. If this is the only place it's used, then let's leave it as an Enum. If we have to copy/paste it anywhere else, better to convert it to a type alias.

**Comment by bishopbm1** [comment]
I added examples to the readme and classes. Should be good to go.

**Pull request merged**